File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/Mul.java
package org.uva.sea.ql.ast.expr;

import org.uva.sea.ql.ast.form.Context;
import org.uva.sea.ql.ast.visit.Visitor;
import org.uva.sea.ql.type.NumericalType;
import org.uva.sea.ql.type.Type;

public class Mul extends BinaryExpr {
	
	public Mul(Expr lhs, Expr rhs) {
		super.lhs = lhs;
		super.rhs = rhs;
	}
	
	@Override
	public Integer eval() {
		return (Integer) lhs.eval() * (Integer) rhs.eval();
	}
	
	public void accept(Visitor visitor, Object context) {
		visitor.visit(this, context);
	}

	@Override
	public String toString() {
		return super.toString() + " (*)";
	}
	
	@Override
	public Type getType(Context context) {
		return new NumericalType();
	}
}

File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/mul.go
package binaryoperatorexpr

import "ql/ast/expr"

type Mul struct {
	BinaryOperator
}

func NewMul(lhs expr.Expr, rhs expr.Expr) Mul {
	return Mul{BinaryOperator{Lhs: lhs, Rhs: rhs}}
}

func (m Mul) Eval(s interface{}) interface{} {
	return m.Lhs.Eval(s).(int) * m.Rhs.Eval(s).(int)
}
